src/components/Sidebar/Sidebar.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+  
+<aside 
+
+  class="w-56 absolute inset-y-0 left-0 transform -translate-x-full transition duration-200 ease-in-out  "
+  :class="{'relative -translate-x-0':showSideBar}"
+  aria-label="Sidebar">
+  <div class="overflow-y-auto py-4 px-3  bg-gray-50  h-screen dark:bg-gray-800">
+    <div class="flex items-center justify-between p-2 tracking-widest text-base font-bold text-gray-900 rounded-lg dark:text-white ">
+      <p class="p-[10px]">
+        BeforeClass
+      </p>
+    <button class="hover:bg-gray-100 dark:hover:bg-gray-700 p-2 rounded-lg" @click="showSideBar = !showSideBar">
+      <svg class="flex-shrink-0 w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white " fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M11 19l-7-7 7-7m8 14l-7-7 7-7"></path></svg>
+    </button>
+    </div>
+      <ul class="space-y-2 p-[10px] mt-[15px]">
+        <li>
+          <a href="#" class="flex items-center p-2 text-base font-normal text-gray-900 rounded-lg dark:text-white hover:bg-gray-100 dark:hover:bg-gray-700">
+            <svg aria-hidden="true" class="w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="M2 10a8 8 0 018-8v8h8a8 8 0 11-16 0z"></path><path d="M12 2.252A8.014 8.014 0 0117.748 8H12V2.252z"></path></svg>
+            <span class="ml-3">Timetable</span>
+          </a>
+        </li>
+        <li>
+          <a href="#" class="flex items-center p-2 text-base font-normal text-gray-900 rounded-lg dark:text-white hover:bg-gray-100 dark:hover:bg-gray-700">
+            <svg class="flex-shrink-0 w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M19 11H5m14 0a2 2 0 012 2v6a2 2 0 01-2 2H5a2 2 0 01-2-2v-6a2 2 0 012-2m14 0V9a2 2 0 00-2-2M5 11V9a2 2 0 012-2m0 0V5a2 2 0 012-2h6a2 2 0 012 2v2M7 7h10"></path></svg>
+            <span class="flex-1 ml-3 whitespace-nowrap">Modules</span>
+          </a>
+        </li>
+      </ul>
+      <ul class="space-y-2 p-[10px] mt-[15px] border-t border-gray-200 dark:border-gray-700">
```

Comment:
not a fan of doing it this way but it looks fine aesthetically

if you can think of a way to render a divider line by itself rather than using borders of an unordered list that would be ideal, but otherwise this implementation should be fine as well